drag_and_drop_v2/utils.py

```diff
@@ -78,3 +78,139 @@ def not_placed(number, ngettext=ngettext_fallback):
+        """
+        Applies zone migrations
+        """
+        migrations = (cls._zone_v1_to_v2, cls._zone_v2_to_v2p1)
+
+        return cls._apply_migration(zone, migrations)
+
+    @classmethod
+    def apply_item_state_migrations(cls, item_state):
+        """
+        Applies item_state migrations
+        """
+        migrations = (cls._item_state_v1_to_v1p5, cls._item_state_v1p5_to_v2, cls._item_state_v1p5_to_v2p1)
+
+        return cls._apply_migration(item_state, migrations)
+
+    @classmethod
+    def _zone_v1_to_v2(cls, zone):
+        """
+        Migrates zone data from v1.0 format to v2.0 format.
+
+        Changes:
+        * v1 used zone "title" as UID, while v2 zone has dedicated "uid" property
+        * "id" and "index" properties are no longer used
+
+        In: {'id': 1, 'index': 2, 'title': "Zone", ...}
+        Out: {'uid': "Zone", ...}
+        """
+        if "uid" not in zone:
+            zone["uid"] = zone.get("title")
+        zone.pop("id", None)
+        zone.pop("index", None)
+
+        return zone
+
+    @classmethod
+    def _zone_v2_to_v2p1(cls, zone):
+        """
+        Migrates zone data from v2.0 to v2.1
+
+        Changes:
+        * Removed "none" zone alignment; default align is "center"
+
+        In: {
+            'uid': "Zone", "align": "none",
+            "x_percent": "10%", "y_percent": "10%", "width_percent": "10%", "height_percent": "10%"
+        }
+        Out: {
+            'uid': "Zone", "align": "center",
+            "x_percent": "10%", "y_percent": "10%", "width_percent": "10%", "height_percent": "10%"
+        }
+        """
+        if zone.get('align', None) not in Constants.ALLOWED_ZONE_ALIGNMENTS:
+            zone['align'] = Constants.DEFAULT_ZONE_ALIGNMENT
+
+        return zone
+
+    @classmethod
+    def _item_state_v1_to_v1p5(cls, item):
+        """
+        Migrates item_state from v1.0 to v1.5
+
+        Changes:
+        * Item state is now a dict instead of tuple
+
+        In: ('100px', '120px')
+        Out: {'top': '100px', 'left': '120px'}
+        """
+        if isinstance(item, dict):
+            return item
+        else:
+            return {'top': item[0], 'left': item[1]}
+
+    @classmethod
+    def _item_state_v1p5_to_v2(cls, item):
+        """
+        Migrates item_state from v1.5 to v2.0
+
+        Changes:
+        * Item placement attributes switched from absolute (left-top) to relative (x_percent-y_percent) units
+
+        In: {'zone': 'Zone", 'correct': True, 'top': '100px', 'left': '120px'}
+        Out: {'zone': 'Zone", 'correct': True, 'top': '100px', 'left': '120px'}
+        """
+        # Conversion can't be made as parent dimensions are unknown to python - converted in JS
+        # Since 2.1 JS this conversion became unnecesary, so it was removed from JS code
+        return item
+
+    @classmethod
+    def _item_state_v1p5_to_v2p1(cls, item):
+        """
+        Migrates item_state from v1.5 to v2.5
```

Comment:
@haikuginger Following up on https://github.com/edx-solutions/xblock-drag-and-drop-v2/pull/92#discussion-diff-74210508, this should be `v2.0` to `v.2.1`.